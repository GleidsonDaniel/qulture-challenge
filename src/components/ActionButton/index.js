import React, {memo} from 'react';
import PropTypes from 'prop-types';
import {FloatingAction} from 'react-native-floating-action';
import Icon from 'react-native-vector-icons/FontAwesome';

import {qultureGreen, baseWhite} from '~/helpers/colors';

const ActionButton = ({actions, onPress, icon, color}) => {
  return (
    <FloatingAction
      onPressMain={onPress}
      color={color}
      floatingIcon={<Icon name={icon} size={30} color={baseWhite} />}
      actions={actions}
      showBackground={false}
    />
  );
};

ActionButton.propTypes = {
  actions: PropTypes.arrayOf(PropTypes.object),
  onPress: PropTypes.func,
  icon: PropTypes.string,
  color: PropTypes.string,
};

ActionButton.defaultProps = {
  actions: [],
  onPress: () => {},
  icon: 'pencil',
  color: qultureGreen,
};

export default memo(ActionButton);
