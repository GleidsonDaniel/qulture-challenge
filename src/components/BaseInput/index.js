import React, {memo} from 'react';
import PropTypes from 'prop-types';

import {Input, TextError, TextLabel, Container} from './styles';

const BaseInput = ({
  label,
  onSubmitEditing,
  onChangeText,
  error,
  errorMessage,
}) => {
  return (
    <Container>
      {!!label && <TextLabel>{label}</TextLabel>}
      <Input onSubmitEditing={onSubmitEditing} onChangeText={onChangeText} />
      {!!error && <TextError>{errorMessage}</TextError>}
    </Container>
  );
};

BaseInput.propTypes = {
  label: PropTypes.string,
  onSubmitEditing: PropTypes.func,
  onChangeText: PropTypes.func,
  error: PropTypes.bool,
  errorMessage: PropTypes.string,
};

BaseInput.defaultProps = {
  label: '',
  onSubmitEditing: () => {},
  onChangeText: () => {},
  error: false,
  errorMessage: '',
};

export default memo(BaseInput);
