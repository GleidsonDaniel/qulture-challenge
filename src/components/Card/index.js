import React from 'react';
import {View, StyleSheet, Platform, TouchableOpacity} from 'react-native';
import PropTypes from 'prop-types';

import {
  widthPercentageToDP,
  heightPercentageToDP,
} from '~/helpers/pixelHandler';

const Card = ({
  children,
  elevation,
  opacity,
  cornerRadius,
  backgroundColor,
  onPress,
}) => {
  const cardStyle = Platform.select({
    ios: () =>
      StyleSheet.create({
        card: {
          shadowRadius: elevation,
          shadowOpacity: opacity,
          shadowOffset: {width: 0, height: elevation},
          borderRadius: cornerRadius,
          backgroundColor,
          width: widthPercentageToDP('46%'),
          flex: 1,
          alignItems: 'center',
          justifyContent: 'center',
        },
      }),
    android: () =>
      StyleSheet.create({
        card: {
          elevation,
          borderRadius: cornerRadius,
          backgroundColor,
          width: widthPercentageToDP('46%'),
          height: heightPercentageToDP('28%'),
          flex: 1,
          alignItems: 'center',
          justifyContent: 'center',
        },
      }),
  })();

  return (
    <TouchableOpacity style={{alignItems: 'center'}} onPress={onPress}>
      <View style={cardStyle.card}>{children}</View>
    </TouchableOpacity>
  );
};

Card.propTypes = {
  backgroundColor: PropTypes.string,
  elevation: PropTypes.number,
  cornerRadius: PropTypes.number,
  opacity: PropTypes.number,
  onPress: PropTypes.func,
  children: PropTypes.element,
};

Card.defaultProps = {
  backgroundColor: '#ffffff',
  elevation: 3,
  cornerRadius: 5,
  opacity: 0.5,
  onPress: () => {},
  children: <View />,
};

export default Card;
