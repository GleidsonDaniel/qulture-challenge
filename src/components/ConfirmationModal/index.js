import React from 'react';
import {ActivityIndicator} from 'react-native';
import PropTypes from 'prop-types';
import Modal from 'react-native-modal';
import Icon from 'react-native-vector-icons/FontAwesome';

import {qultureGreen} from '~/helpers/colors';

import {
  ModalContainer,
  Container,
  LoadingText,
  ConfirmText,
  ConfirmButtom,
  OkText,
} from './styles';

const ConfirmationModal = ({
  modalVisible,
  loading,
  confirmMessage,
  confirmationCallback,
}) => {
  const handleRender = () => {
    if (loading) {
      return (
        <Container>
          <ActivityIndicator color={qultureGreen} size={100} />
          <LoadingText>Registrando...</LoadingText>
        </Container>
      );
    }
    return (
      <Container>
        <Icon
          name="check-circle-outline"
          color="#7f9928"
          size={100}
          style={{marginBottom: 5}}
        />
        <ConfirmText numberOfLines={1}>{confirmMessage}</ConfirmText>
        <ConfirmButtom onPress={() => modalVisible && confirmationCallback()}>
          <OkText>OK</OkText>
        </ConfirmButtom>
      </Container>
    );
  };

  return (
    <Modal isVisible={modalVisible}>
      <ModalContainer>{handleRender()}</ModalContainer>
    </Modal>
  );
};

ConfirmationModal.propTypes = {
  modalVisible: PropTypes.bool,
  loading: PropTypes.bool,
  confirmMessage: PropTypes.string,
  confirmationCallback: PropTypes.func,
};

ConfirmationModal.defaultProps = {
  modalVisible: false,
  loading: true,
  confirmMessage: '',
  confirmationCallback: () => {},
};

export default ConfirmationModal;
