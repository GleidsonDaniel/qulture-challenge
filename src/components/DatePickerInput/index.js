import React, {memo, useState} from 'react';
import PropTypes from 'prop-types';
import DateTimePicker from '@react-native-community/datetimepicker';
import moment from 'moment';

import {DateButton, TextError, TextLabel, Container, DateText} from './styles';

const DatePickerInput = ({
  label,
  error,
  errorMessage,
  value,
  selectDate,
  editable,
}) => {
  const [showDatePicker, setShowDatePicker] = useState(false);
  return (
    <Container>
      {showDatePicker && (
        <DateTimePicker
          value={new Date()}
          is24Hour
          display="default"
          onChange={(event, date) => {
            if (date) selectDate(moment(date).format('YYYY-MM-DD'));
            setShowDatePicker(false);
          }}
        />
      )}
      {!!label && <TextLabel>{label}</TextLabel>}
      <DateButton disabled={editable} onPress={() => setShowDatePicker(true)}>
        <DateText>{value || 'Selecione aqui a data de admiss√£o'}</DateText>
      </DateButton>
      {!!error && <TextError>{errorMessage}</TextError>}
    </Container>
  );
};

DatePickerInput.propTypes = {
  label: PropTypes.string,
  error: PropTypes.bool,
  errorMessage: PropTypes.string,
  value: PropTypes.string,
  selectDate: PropTypes.func,
  editable: PropTypes.bool,
};

DatePickerInput.defaultProps = {
  label: '',
  error: false,
  errorMessage: '',
  value: '',
  selectDate: () => {},
  editable: true,
};

export default memo(DatePickerInput);
