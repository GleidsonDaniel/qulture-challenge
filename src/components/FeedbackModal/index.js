import React from 'react';
import {Text, ActivityIndicator} from 'react-native';
import Modal from 'react-native-modal';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import {baseRed, qultureGreen} from '~/helpers/colors';

import {
  Container,
  ContentContainer,
  FeedbackText,
  FeedbackButton,
} from './styles';

export default function FeedbackModal({
  setModal,
  loading,
  error,
  finalMessage,
  okClick,
}) {
  const handleRender = () => {
    if (error) {
      return (
        <ContentContainer>
          <Icon
            name="close-circle-outline"
            color={baseRed}
            size={100}
            style={{marginBottom: 5}}
          />
          <FeedbackText numberOfLines={1}>
            Houve um erro, por favor tente novamente.
          </FeedbackText>
        </ContentContainer>
      );
    }
    if (loading) {
      return (
        <ContentContainer>
          <ActivityIndicator color={qultureGreen} size={100} />
          <Text>Carregando...</Text>
        </ContentContainer>
      );
    }
    return (
      <ContentContainer>
        <Icon
          name="check-circle-outline"
          color={qultureGreen}
          size={100}
          style={{marginBottom: 5}}
        />
        <FeedbackText numberOfLines={1}>{finalMessage}</FeedbackText>
        <FeedbackButton onPress={() => okClick()}>
          <FeedbackText>OK</FeedbackText>
        </FeedbackButton>
      </ContentContainer>
    );
  };
  return (
    <Modal
      isVisible
      onBackdropPress={() => loading === false && setModal()}
      onBackButtonPress={() => loading === false && setModal()}>
      <Container>{handleRender()}</Container>
    </Modal>
  );
}
