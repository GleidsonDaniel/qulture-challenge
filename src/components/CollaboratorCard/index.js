/* eslint-disable camelcase */
import React, {memo} from 'react';
import PropTypes from 'prop-types';

import Card from '../Card';
import {
  Container,
  Avatar,
  Name,
  Email,
  AdmissionDate,
  JobTitle,
} from './styles';

import user from '~/assets/icons/user/user.png';

const CollaboratorCard = ({collaborator}) => {
  const httpRegex = /^(http|https)/;
  const imageVerify = () => {
    if (
      httpRegex.test(collaborator.photo_url) &&
      !collaborator.photo_url.includes('imgur')
    ) {
      return {
        uri: collaborator.photo_url,
      };
    }
    return user;
  };
  return (
    <Card>
      <Container>
        <Avatar source={imageVerify()} resizeMode="cover" />
        <Name>{collaborator.name}</Name>
        <Email>{collaborator.email}</Email>
        <AdmissionDate>{collaborator.admission_date}</AdmissionDate>
        <JobTitle>{collaborator.job_title}</JobTitle>
      </Container>
    </Card>
  );
};

CollaboratorCard.propTypes = {
  collaborator: PropTypes.shape({
    photo_url: PropTypes.string,
    name: PropTypes.string,
    email: PropTypes.string,
    admission_date: PropTypes.string,
    job_title: PropTypes.string,
  }),
};

CollaboratorCard.defaultProps = {
  collaborator: {
    photo_url: '',
    name: '',
    email: '',
    admission_date: '',
    job_title: '',
  },
};

export default memo(CollaboratorCard);
