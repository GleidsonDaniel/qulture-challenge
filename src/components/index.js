import CollaboratorCard from './CollaboratorCard';
import Loading from './Loading';
import BaseInput from './BaseInput';
import ActionButton from './ActionButton';
import FeedbackModal from './FeedbackModal';
import DatePickerInput from './DatePickerInput';

export {
  CollaboratorCard,
  Loading,
  BaseInput,
  ActionButton,
  FeedbackModal,
  DatePickerInput,
};
