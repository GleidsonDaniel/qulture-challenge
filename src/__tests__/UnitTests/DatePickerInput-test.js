import React from 'react';
import {shallow} from 'enzyme';
import {DatePickerInput} from '~/components';

describe('DatePickerInput', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const component = shallow(<DatePickerInput />);
      expect(component).toMatchSnapshot();
    });
  });
});
