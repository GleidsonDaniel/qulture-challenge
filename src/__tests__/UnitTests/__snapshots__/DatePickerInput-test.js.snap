// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePickerInput Rendering should match to snapshot 1`] = `ShallowWrapper {}`;
